### PR TITLE
Upstream RDK adapter

### PR DESCRIPTION
**Pull request by kkanag314**
Upstream the RDK Adapter

### CODE REVIEW

**Comment by mhughesacn** [comment]
Hi Krishna: Please use Metrological copyright, not RDK.
Martin Hughes, RDK CMF Compliance Team

**Review by pwielders — CHANGES_REQUESTED** [review]
From the past, what you might have copied, might be a bit superseded by new functionality. If you check the CMakeList.txt (see: https://github.com/rdkcentral/ThunderClientLibraries/blob/8b6f54c434a1c1fab0b1ce93f56f1a44deb18bcd/Source/ocdm/CMakeLists.txt#L35C45-L36C1) you see that the Mesaging (Tracing/Logging) is included in the build of this module, so suggest tor replace the TRACE_LX macros by the TRACE macro and definititly do *not* use the printf stderr output for multiple reasons:
1) fprintf, can not be turned on/off @ runtime (TRACES can)
2) stdou/stderr can be redirected to /dev/null, so will never be logged
3) If enabled, there is no scynronisation while outputting to the streams (stdout/stderr) and thus message might be scattered through all kinds of other messages.
To circumvent these issue, please use Tracing/Logging as delivered by the Thunder framewrok. See: https://rdkcentral.github.io/Thunder/plugin/messaging/#tracing and https://rdkcentral.github.io/Thunder/plugin/messaging/#logging for more details

**Review by pwielders — CHANGES_REQUESTED** [review]
To avoid breaking backward compatibility, also add the new adapter methods (opencdm_adapter_construct_session/opencdm_adapter_destruct_session) to the existing adapter implementations. These can be straithforward calls tp the base class, but I do see any changes on the existing two adapters.

File: Source/ocdm/adapter/open_cdm_adapter.h [hunk 1]
```diff
@@ -88,6 +88,45 @@ extern "C" {
...
 
+/**
+ * \brief adds SVP related features to the caps structure (only if needed by the platform)
+ *
+ * \param caps GstCaps to be updated
+ *
+ * \return Zero on success, non-zero on error.
+ */
+
+    EXTERNAL OpenCDMError opencdm_gstreamer_transform_caps(GstCaps** caps);
```
Comment (pwielders) [inline Source/ocdm/adapter/open_cdm_adapter.h:99]: I guess this method does have an implementation on the current SVP adapter as well. Why was this not a nessecity in the previus setup? Is this not part of the implementation of the decrypt? What I am trying to say :-) Is this method really nessecary otr is this implementation leakage ? If the latetr, please remove, and move it into the decrypt call

**Pull request closed without merging**